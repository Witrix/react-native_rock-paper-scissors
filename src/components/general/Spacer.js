import React, { Component } from 'react';
import { View }             from 'react-native';

const DEFAULT_SIZE = 20;

export default class Spacer extends Component {
    constructor(props) {
        super(props);
        let size = DEFAULT_SIZE;
        if (props.size && typeof props.size === 'number' && props.size >= 0) size = props.size;
        this.state = {
            size: size,
        }
    }

    render() {
        return (
            <View style={{marginTop: this.state.size || DEFAULT_SIZE}}/>
        );
    }
}
