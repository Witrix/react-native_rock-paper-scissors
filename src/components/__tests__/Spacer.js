import 'react-native';
import React      from 'react';
import Spacer     from '../general/Spacer';
import renderer   from 'react-test-renderer';

it('Render correctly: No props', () => {
  const tree = renderer.create(
    <Spacer />
  );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: Correct props', () => {
    const tree = renderer.create(
        <Spacer size={20}/>
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: Size null', () => {
    const tree = renderer.create(
        <Spacer size={null}/>
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: Size invalid', () => {
    const tree = renderer.create(
        <Spacer size={-1}/>
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: Size invalid', () => {
    const tree = renderer.create(
        <Spacer size={"abc"}/>
    );
    expect(tree).toMatchSnapshot();
});

