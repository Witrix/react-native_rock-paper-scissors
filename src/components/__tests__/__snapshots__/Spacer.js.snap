// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly: Correct props 1`] = `
<View
  style={
    Object {
      "marginTop": 20,
    }
  }
/>
`;

exports[`Render correctly: No props 1`] = `
<View
  style={
    Object {
      "marginTop": 20,
    }
  }
/>
`;

exports[`Render correctly: Size invalid 1`] = `
<View
  style={
    Object {
      "marginTop": 20,
    }
  }
/>
`;

exports[`Render correctly: Size invalid 2`] = `
<View
  style={
    Object {
      "marginTop": 20,
    }
  }
/>
`;

exports[`Render correctly: Size null 1`] = `
<View
  style={
    Object {
      "marginTop": 20,
    }
  }
/>
`;
